diff --git a/lib/module/controllers/LLMController.js b/lib/module/controllers/LLMController.js
index 5bfa8e26a4a33de6e95d843bd5db27878d1a0517..56ca253d16294217d4a996d8bf6934917a86487a 100644
--- a/lib/module/controllers/LLMController.js
+++ b/lib/module/controllers/LLMController.js
@@ -4,7 +4,7 @@ import { ResourceFetcher } from '../utils/ResourceFetcher';
 import { ETError, getError } from '../Error';
 import { Template } from '@huggingface/jinja';
 import { DEFAULT_CHAT_CONFIG } from '../constants/llmDefaults';
-import { readAsStringAsync } from 'expo-file-system';
+import { readAsStringAsync } from 'expo-file-system/legacy';
 import { SPECIAL_TOKENS } from '../types/llm';
 import { parseToolCall } from '../utils/llm';
 import { Logger } from '../common/Logger';
diff --git a/src/controllers/LLMController.ts b/src/controllers/LLMController.ts
index cde0c6480200215439af8ee3b043a1c40979e325..b39b4ed56423f4e5558fb4ae83dba0186253643a 100644
--- a/src/controllers/LLMController.ts
+++ b/src/controllers/LLMController.ts
@@ -3,7 +3,7 @@ import { ResourceFetcher } from '../utils/ResourceFetcher';
 import { ETError, getError } from '../Error';
 import { Template } from '@huggingface/jinja';
 import { DEFAULT_CHAT_CONFIG } from '../constants/llmDefaults';
-import { readAsStringAsync } from 'expo-file-system';
+import { readAsStringAsync } from 'expo-file-system/legacy';
 import {
   ChatConfig,
   GenerationConfig,
